New article "Style editing for CSS-in-JS frameworks" (#1378)
MicrosoftDocs/edge-developer: microsoft-edge/devtools-guide-chromium/css/css-in-js.md

## Blog post

This article talks about CSS-in-JS support in DevTools and, in general, what we mean by _CSS-in-JS_ and how it's different from regular CSS.

## What is CSS-in-JS?

The definition of CSS-in-JS is somewhat vague. In a broad sense, it's an approach for managing CSS code using JavaScript. For example, it could mean that the CSS content is defined using JavaScript and the final CSS output is generated on-the-fly by the app.

In the context of DevTools, _CSS-in-JS_ means that the CSS content is injected into the page using CSSOM APIs.  Regular CSS is injected using `<style>` or `<link>` elements, and it has a static source (such as a DOM node or a network resource).  In contrast, CSS-in-JS often doesn't have a static source.  A special case here is that the content of a `<style>` element can be updated using CSSOM API, causing the source to become out of sync with the actual CSS stylesheet.

If you use any CSS-in-JS library (such as styled-component, Emotion, or JSS), the library might inject styles using CSSOM APIs under the hood depending on the mode of development and the browser.

> Let's look at some examples on how you can inject a stylesheet using CSSOM API similar to what CSS-in-JS libraries are doing.

Reviewer: ```suggestion
Let's look at some examples on how you can inject a stylesheet using CSSOM API similar to how CSS-in-JS libraries work.
```

Reviewer: Ending a sentence with "doing" just sounds weird ... it could just end with "similar to CSS-in-JS libraries." or "in the same manner as CSS-in-JS libraries."  I just made a suggestion to start the conversation.
